### ISSUE TITLE
man2lore doesn't correctly translate .TP tags

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3017]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#3017|
|Type|enhancement|
|Created|2008-01-31 15:53:15Z|
|Branch|https://github.com/twisted/twisted/tree/fix-man2lore-3017|

Basically, it means that the output of the list of options in xhtml is broken.

<details><summary>Searchable metadata</summary>

```
trac-id__3017 3017
type__enhancement enhancement
reporter__therve therve
priority__highest highest
milestone__ 
branch__branches_fix_man2lore_3017 branches/fix-man2lore-3017
branch_author__ 
status__closed closed
resolution__fixed fixed
component__lore lore
keywords__ 
time__1201794795000000 1201794795000000
changetime__1202114383000000 1202114383000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [22432]) Branching to 'fix-man2lore-3017'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [22433]) Fix and tests.

Refs [#3017](https://github.com/twisted/twisted/issues/3017)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

This is ready to review. man2lore would need one hundred time more tests, but I'm not sure I have the energy to do that right now. As it's a release blocker...

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @therve|
|-|-|

eyes bleeding, etc.  Thanks for tackling this.  I won't ask for all the extra tests this code really needs, since the change to fix this issue is pretty small and the approach `man2lore` takes is _completely_ wrong anyway...

Some stuff I think should be done, though:

  * test-case-name for man2lore.py
  * docstring for `macro_TP` and `text` methods of `ManConverter`, since they had changes
  * documentation for the `tp` ivar of `ManConverter`
  * Factor the common parts of the tests into a helper - writing to a file, calling convert, etc.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [22439]) Process review comments.

Refs [#3017](https://github.com/twisted/twisted/issues/3017).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve removed owner|
|-|-|

Thanks for the review. I think I fix everything, except the test-case-name which was already there :).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @therve|
|-|-|

Cool, thanks.  `ManConverter.tp` is still pretty opaque to me though.  Do you think you could document what each of the 4 values it can take on mean (I ask because I think you must have figured out at least some of them in order to make this change :)?  This could either take the form of an expanded `@ivar` or maybe the constants could be replaced with symbolic names which have some meaning.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve commented|
|-|-|

(In [22459]) Document tp variable.

Refs [#3017](https://github.com/twisted/twisted/issues/3017)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve removed owner|
|-|-|

I choosed the verbose way, I hope this is clear enough. Back to review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @therve|
|-|-|

Looks good, thanks.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set status to `closed`|
|-|-|

(In [22460]) Merge fix-man2lore-3017

Author: therve
Reviewer: exarkun
Fixes [#3017](https://github.com/twisted/twisted/issues/3017)

Fix the conversion of the list of options in man pages to Lore format.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
